midend/interpreter.cpp

```diff
@@ -859,6 +859,16 @@ void ExpressionEvaluator::postorder(const IR::Operation_Unary* expression) {
         return;
     }
     if (sv->isUnknown()) {
+        if (auto cast = expression->to<IR::Cast>()) {
+            if (cast->destType->is<IR::Type_Boolean>() &&
+                l->is<SymbolicInteger>()) {
+                l = new SymbolicBool(sv->state);
+
+            } else if (cast->destType->is<IR::Type_Bits>() &&
+                       l->is<SymbolicBool>()) {
+                l = new SymbolicInteger(sv->state, cast->destType->to<IR::Type_Bits>());
+            }
```

Comment:
ok, then how about an additional check here: `else BUG("%1% unexpected type %2% in cast", cast, cast->destType)`.
This would ensure that the pass is "safe" even if someone changes the order of the midend passes. It will still crash the compiler, but it will give a somewhat clearer explanation.